File: Frontend/src/events/converters/battle.converter.ts
import type { BattleDTO } from "@/events/dto/battle.dto";
import type { Battle } from "@/models/battle";
import { EnemyConverter } from "./battles/enemy.converter";

export class BattleConverter {
  public static convert(battleDto: Required<BattleDTO>): Battle {
    return {
      field: battleDto.field ?? 0,
      groupId: battleDto.groupId ?? 0,
      enemy: EnemyConverter.convert(battleDto.enemy ?? null),
    };
  }
}

File: Backend/Events/Converters/BattleConverter.cs
using Backend.Domain.Models;
using Backend.Events.Converters.Battles;
using Backend.Events.DTO;

namespace Backend.Events.Converters;

public static class BattleConverter
{
    public static BattleDTO ToDTO(Battle battle) => new()
    {
        Field = battle.Field,
        GroupId = battle.GroupId,
        Enemy = EnemyConverter.ToDTO(battle.Enemy)
    };
}
